source/dub/packagesupplier.d

```diff
@@ -290,6 +290,150 @@ class RegistryPackageSupplier : PackageSupplier {
 	}
 }
 
+/**
+	Maven repository based package supplier.
+
+	This package supplier connects to a maven repository
+	to search for available packages.
+*/
+class MavenRegistryPackageSupplier : PackageSupplier {
+	private {
+		URL m_mavenUrl;
+		struct CacheEntry { Json data; SysTime cacheTime; }
+		CacheEntry[string] m_metadataCache;
```

Comment:
When only new packages are added this cache is never cleared. It doubt it matters because usage is through one-time CLI mostly + there are less than a million MVN packages.